### DISCUSSION TITLE
Authenticated request?

### DISCUSSION THREAD

**Discussion by vonec** [dedoc/scramble#6] [Q&A] [answered]
is there any way we can add @authenticated and json tokens to the UI?

**Comment by romalytvynenko** [comment]
Hey @vonec!
Yeah, it should be possible. Can you tell me a bit more about your application? As I understood, you have endpoints which require auth and you use JWT tokens for that. Is that correct?

**Comment by romalytvynenko** [comment, marked as answer]
Hey @vonec 

To describe your API authentication, you can extend you OpenAPI document and add the security scheme to it.

```php
Scramble::extendOpenApi(function (OpenApi $openApi) {
    $openApi->secure(
        SecurityScheme::http('bearer', 'JWT')
    );
});
```

Read more about security here: https://scramble.dedoc.co/usage/security

**Reply by vonec** [reply]
Thank you ❤️

**Reply by vonec** [reply]
Upgraded the package to 0.5.0 and it broke the API.json with error type

**Reply by vonec** [reply]
TypeError: Return value of Dedoc\Scramble\Support\InferExtensions\JsonResourceTypeInfer::modelType() must implement interface Dedoc\Scramble\Support\Type\Type, null returned in file \vendor\dedoc\scramble\src\Support\InferExtensions\JsonResourceTypeInfer.php on line 120

can you suggest me what didnt resolve

**Reply by romalytvynenko** [reply]
@vonec should've been fixed in 0.7.2

Feel free to re-open if you have any questions.
